src/layers/conv.jl

```diff
@@ -118,29 +121,27 @@ function (l::GCNConv)(g::GNNGraph,
         end
     end
     Dout, Din = size(l.weight)
-    if Dout < Din
-        # multiply before convolution if it is more convenient, otherwise multiply after
-        x = l.weight * x
-    end
-    if edge_weight !== nothing
-        d = degree(g, T; dir = :in, edge_weight)
+
+    if g isa GNNHeteroGraph
+        d = degree(g, edge_t; dir = :in)
     else
-        d = degree(g, T; dir = :in, edge_weight = l.use_edge_weight)
+        if edge_weight !== nothing
+            d = degree(g; dir = :in, edge_weight)
```

Comment:
Degree should receive an eltype `T` as well. Probably the CI failure is due to `d` here having eltype `Int`
Then `1 ./ sqrt.(d)` is a float64 vector and this propagates to the final output

Reply:
T is not defined now even in the function signature, as we need to remove that constraint for Heterograph.
Also, default for T is nothing, and when `T = nothing` is passed to the degree function, type should be inferred from the type of `src`. I think it should be inferred from the feature matrix instead?

Reply:
You can define T = eltype(xj)

Reply:
I know, just did it 7 minutes ago, waiting for tests :wheel_of_dharma: